source/language/types.rst

```diff
@@ -268,6 +268,56 @@ are all of type ``float[64]``.
 
 Note that `e` is a valid identifier. `e/E` are also used in scientific notation where appropriate.
 
+Arrays
+------
+
+Statically-sized arrays of values can be created and initialized, and individual elements
+can be accessed, using the following general syntax:
+
+.. code-block:: c
+
+   array[int[32], 5] myArray = {0, 1, 2, 3, 4};
+   array[float[32], 3, 2] multiDim = {{1.1, 1.2}, {2.1, 2.2}, {3.1, 3.2}};
+
+   int[32] firstElem = myArray[0]; // 0
+   int[32] lastElem = myArray[4]; // 4
+   int[32] alsoLastElem = myArray[-1]; // 4
+   float[32] firstLastElem = multiDim[0][1]; // 1.2
+   float[32] lastLastElem = multiDim[2][1]; // 3.2
+   float[32] alsoLastLastElem = multiDim[-1][-1]; // 3.2
+
+   myArray[4] = 10; // myArray == {0, 1, 2, 3, 10}
+   multiDim[0][0] = 0.0; // multiDim == {{0.0, 1.2}, {2.1, 2.2}, {3.1, 3.2}}
+   multiDim[-1][1] = 0.0; // multiDim == {{0.0, 1.2}, {2.1, 2.2}, {3.1, 0.0}}
+
+Arrays *cannot* be declared inside the body of a function or gate. All arrays
+*must* be declared within the global scope of the program.
+Indexing of arrays is n-based *i.e.*, negative indices are allowed.
+Multi-dimensional arrays (as in the example above) are allowed, with a maximum
+of 7 total dimensions.
+
+For backwards compatability, the standard
```

Comment:
For 'interoperability' with quantum registers, etc.? ("Backwards compatibility" suggests that this syntax was present in OpenQASM2, which it wasn't.)

Reply:
Changed in https://github.com/Qiskit/openqasm/pull/283/commits/7908b8aa4ebebabb4a58e256b4c5216951a6f2e7